### DISCUSSION TITLE
Infinite loop error using laravel-nestedset package

### DISCUSSION THREAD

**Discussion by rezasm** [filamentphp/filament#7050] [Help]
I am building a nested category manager in filament using laravel-nestedset package.
when I use `toTree()` method I get this error:

**Xdebug has detected a possible infinite loop, and aborted your script with a stack depth of '256' frames**

this is the code in the widget:

```php
<?php

namespace App\Filament\Widgets;

use Book\Models\Category;
use Filament\Widgets\Widget;

class CategoryTree extends Widget
{
    protected static string $view = 'filament.widgets.category-tree';

    protected function getViewData(): array
    {

        return [
            'categories' => Category::get()->toTree()
        ];
    }

}
```

`toTree()` is a method from laravel-nestedset package.

and this is the widget blade code:
```blade
<x-filament::widget>
    <x-filament::card>


        <ul class="nested" id="items">
            @livewire('sortable-list',['cats' => $categories])
        </ul>
 
    </x-filament::card>
</x-filament::widget>

```
this only happens in Filament. 

I tested it in a blade file the issue resolved when I moved `$cats` property to the `render()` method:
```php
  public function render()
    {
        $cats  = Category::get()->toTree();
         return view('livewire.test-tree',['cats' => $cats]);
    }
```
 but in Filament the issue still remains.

**Comment by zipavlin** [comment]
@rezasm did you maybe manage to figure this out? I would also like to implement nestedset (or similar) for page tree.
